diff-analyzer/src/main/java/org/starchartlabs/chronicler/diff/analyzer/PullRequestAnalyzer.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) Sep 26, 2018 StarChart Labs Authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    romeara - initial API and implementation and/or initial documentation
+ */
+package org.starchartlabs.chronicler.diff.analyzer;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.starchartlabs.chronicler.calamari.core.auth.InstallationAccessToken;
+import org.starchartlabs.chronicler.events.GitHubPullRequestEvent;
+import org.starchartlabs.chronicler.github.model.PageReader;
+import org.starchartlabs.chronicler.github.model.Requests;
+import org.starchartlabs.chronicler.github.model.pullrequest.StatusHandler;
+
+import com.google.gson.JsonElement;
+
+import okhttp3.HttpUrl;
+
+//https://developer.github.com/v3/pulls/#list-pull-requests-files
+public class PullRequestAnalyzer {
+
+    /** Logger reference to output information to the application log files */
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final Supplier<String> applicationKey;
+
+    public PullRequestAnalyzer(Supplier<String> applicationKey) {
+        this.applicationKey = Objects.requireNonNull(applicationKey);
+    }
+
+    public void analyze(GitHubPullRequestEvent event) {
+        logger.info("Processing pull request for {}", event.getLoggableRepositoryName());
+
+        InstallationAccessToken accessToken = InstallationAccessToken.forRepository(event.getBaseRepositoryUrl(),
+                applicationKey, Requests.USER_AGENT);
+
+        StatusHandler statusHandler = new StatusHandler("doc/chronicler", event.getPullRequestStatusesUrl(),
+                accessToken);
+
+        // Set pending status
+        statusHandler.sendPending("Analysis in progress");
+
+        AnalysisSettings settings = AnalysisSettings.defaultSettings();
+
+        HttpUrl url = HttpUrl.get(event.getPullRequestUrl()).newBuilder()
+                .addPathSegment("files")
+                .addQueryParameter("page", "1")
+                .addQueryParameter("per_page", "30")
+                .build();
+
+        try {
+            // TODO test new paging
```

Comment:
have you written this test?

Reply:
No, I am going to do a bit more tinkering before I add tests to all the stuff - don't want to take time for unit tests and such until I'm more sure of the final form of the classes